### PR TITLE
Evaluation of bergamot translator

### PR DESCRIPTION
**Pull request by eu9ene**
What it does:
- prepares reproducible docker environment
- compiles `bergamot-translator` and `marian-dev`
- downloads prod Mozilla models
- calculates BLEU scores using SacreBLEU and official WMT datasets
- automatically discovers which language pairs are supported using Mozilla models
- automatically discovers WMT datasets for supported language pairs
- calls Google and Microsoft translation APIs 
- builds a report to consolidate and compare evaluation results for Bergamot, Marian, Microsoft and Google 
- uses the same settings for `bergamot-translator` as used in the extension
- uses unified API of `bergamot-translator`, the same which is used from WASM in the extension (evaluation app: https://github.com/browsermt/bergamot-translator/pull/76)

### CODE REVIEW

**Review by andrenatal — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# pcre2 is requried to build berrgamot-translator with -DUSE_WASM_COMPATIBLE_SOURCES=off
+RUN apt-get install -y libpcre2-dev
+
+# SacreBLEU and python dependencies
+RUN apt-get update && apt-get install -y python3 python3-venv python3-pip
+ADD ./requirements.txt ./
+RUN pip3 install -r requirements.txt
+
+
+CMD ["/bin/bash"]
```
Comment (motin) [inline Dockerfile:32]: Nit: Ending files with newline makes diffs easier to read after file is appended in future revisions

File: eval/eval.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+mkdir -p $EVAL_DIR
+mkdir -p $EVAL_DIR/$SRC-$TRG
+
+sacrebleu -t $DATASET -l $SRC-$TRG --echo src \
+    | tee $EVAL_DIR/$SRC-$TRG/$DATASET.$SRC \
+    | $TRANSLATOR_CMD \
+    | tee $EVAL_DIR/$SRC-$TRG/$DATASET.$TRANSLATOR.$TRG \
+    | sacrebleu --score-only -q -t $DATASET -l $SRC-$TRG \
+    | tee $EVAL_DIR/$SRC-$TRG/$DATASET.$TRANSLATOR.$TRG.bleu
```
Comment (motin) [inline eval/eval.sh:15]: Nit: Ending files with newline makes diffs easier to read after file is appended in future revisions

File: install/download-models.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    echo "Already cloned"
+else
+    echo "Cloning https://github.com/mozilla-applied-ml/bergamot-models.git"
+    git clone https://github.com/mozilla-applied-ml/bergamot-models.git
+fi
+
+cd bergamot-models/prod
+
+echo "Extracting models"
+gzip -d -r *
```
Comment (motin) [inline install/download-models.sh:17]: Nit: Ending files with newline makes diffs easier to read after file is appended in future revisions

File: install/install-bergamot-translator.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    git clone https://github.com/mozilla/bergamot-translator.git
+fi
+
+cd bergamot-translator
+
+echo "Compiling bergamot-translator"
+mkdir -p build
+cd build
+cmake ../ -DUSE_WASM_COMPATIBLE_SOURCES=off -DCMAKE_BUILD_TYPE=Release
+make -j$(nproc)
```
Comment (motin) [inline install/install-bergamot-translator.sh:23]: Nit: Ending files with newline makes diffs easier to read after file is appended in future revisions

**Review by motin — APPROVED** [review]
Great initial setup! I left mostly nits and typo comments, nothing that prevents merging.

File: translators/microsoft.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    target = os.environ['TRG']
+
+    params = {
+        'api-version': '3.0',
+        'from': source,
+        'to': [target]
+    }
+
+    results = []
+    # decrease batch size if hitting limit of max 10000 characters per request
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply translators/microsoft.py]: it refers to the partition, changed it

**Pull request merged**